SU2_CFD/include/numerics_structure.hpp

```diff
@@ -2865,138 +2897,168 @@ class CAvgGradArtComp_Flow : public CNumerics {
 };
 
 /*!
- * \class CAvgGrad_TurbSA
- * \brief Class for computing viscous term using average of gradients (Spalart-Allmaras Turbulence model).
+ * \class CAvgGrad_Abstract
+ * \brief Template class for computing viscous residual of scalar values
+ * \details This class serves as a template for the scalar viscous residual
+ *   classes.  The general structure of a viscous residual calculation is the
+ *   same for many different  models, which leads to a lot of repeated code.
+ *   By using the template design pattern, these sections of repeated code are
+ *   moved to a shared base class, and the specifics of each model
+ *   are implemented by derived classes.  In order to add a new residual
+ *   calculation for a viscous residual, extend this class and implement
+ *   the pure virtual functions with model-specific behavior.
  * \ingroup ViscDiscr
- * \author A. Bueno.
+ * \author C. Pederson, A. Bueno, and F. Palacios
  * \version 5.0.0 "Raven"
  */
-class CAvgGrad_TurbSA : public CNumerics {
-private:
-  
-  su2double **Mean_GradTurbVar;
-  su2double *Proj_Mean_GradTurbVar_Kappa, *Proj_Mean_GradTurbVar_Edge;
-  su2double *Edge_Vector;
+class CAvgGrad_Abstract : public CNumerics {
```

Comment:
I would suggest to use a different name for this class. Simply CAvgGrad_Turb would be ok I think.